docs/naps/7-key-binding-dispatch.md

```diff
@@ -0,0 +1,427 @@
+
+In case (B), the corresponding command will never be triggered so long as it indirectly conflicts with another key binding. In this sense, multi-part key bindings will always take priority over single-part key bindings.
+
+### Finding a match
+
+When checking if an active key binding matches the entered key sequence, the resolver will fetch the pre-sorted list of direct conflicts and check if the last entry is active using its `when` property, moving to the next entry if it is not. When it encounters a blocking rule, it will return no match, and for a negate rule, it will store the affected command in an ignore list and continue to the next entry. If no special rules are present, it will return a match if the command is not in an ignore list, otherwise continuing to the next entry, and so on, until no more entries remain.
+
+In psuedo-code this reads as:
+```python
+def find_active_match(entries: List[KeyBindingEntry]) -> Optional[KeyBindingEntry]:
+    ignored_commands = []
+
+    for entry in reversed(entries):
+        if isactive(entry.when):
+            if entry.block_rule:
+                return None
+            elif entry.negate_rule:
+                ignored_commands.append(entry.command_id[1:])
+            elif entry.command_id not in ignored_commands:
+                return entry
+```
+
+### Lookup and partial matches
+
+Key bindings can be stored in a map in integer form, as `KeyMod`, `KeyCode`, `KeyCombo`, and `KeyChord` are all represented as unique `int`s with 16 bits per part:
+```python
+keymap = Dict[int, List[KeyBindingEntry]] = {
+    KeyMod.CtrlCmd | KeyCode.KeyZ: ...,
+    KeyMod.CtrlCmd | KeyMod.Shift | KeyCode.KeyZ: ...,
+    KeyMod.CtrlCmd | KeyCode.KeyX: ...,
+    KeyChord(KeyMod.CtrlCmd | KeyCode.KeyX, KeyCode.KeyC): ...,
+    KeyChord(KeyMod.CtrlCmd | KeyCode.KeyX, KeyCode.KeyV): ...,
+    KeyMod.Shift : ...,
+}
+```
+
+To determine if an indirect conflict is present, entries would be filtered via bitwise operations:
+```python
+def has_shift(key: int) -> bool:
+    return bool(key & KeyMod.Shift)
+
+def starts_with_ctrl_cmd_x(key: int) -> bool:
+    return key & 0x0000FFFF == (KeyMod.CtrlCmd | KeyCode.KeyX)
+
+def multi_part(key: int) -> bool:
+    return key > 0x0000FFFF
+
+> list(filter(has_shift, keymap))
+[<KeyCombo.CtrlCmd|Shift|KeyZ: 3115>, <KeyMod.Shift: 1024>]
+
+> list(filter(starts_with_ctrl_cmd_x, keymap))
+[
+    <KeyCombo.CtrlCmd|KeyX: 2089>,
+    KeyChord(<KeyCombo.CtrlCmd|KeyX: 2089>, <KeyCode.KeyC: 20>),
+    KeyChord(<KeyCombo.CtrlCmd|KeyX: 2089>, <KeyCode.KeyV: 39>),
+]
+
+> list(filter(multi_part, keymap))
+[
+    KeyChord(<KeyCombo.CtrlCmd|KeyX: 2089>, <KeyCode.KeyC: 20>),
+    KeyChord(<KeyCombo.CtrlCmd|KeyX: 2089>, <KeyCode.KeyV: 39>),
+]
+```
```

Comment:
The formatting here is a bit weird... I suggest not mixing pure Python cells and cells containing prompts and outputs.